### ISSUE TITLE
selected option does not update when options is changed dynamically

### ISSUE THREAD

**Issue by Noaspz** [storybookjs/storybook#10286]
I've noticed a bug in storybook knobs.
I have one knob that is responsible for the page background color. It selects the background color out of a close group of colors. It looks like this:

`select('Background Color', colors, colors.first, ‘Background’);`

The second knob is responsible for the group of colors:

```js
const colors = {
  first: color('First', '#000000', 'Color Picker’),
  second: color('Second', '#9E9E9E', 'Color Picker'),
  third: color('Third', '#FFFFFF', 'Color Picker'),
};
```

The problem is, for example, when ‘first’ color is selected as a background color, which means the background color is now black. I want to change the ‘first’ color using the ‘Color Picker’ to a different color ( from black to pink for example…).
When I’m changing the ‘first’ color, the background color does not immediately changed. Which means, even though I changed the ‘first’ color from black to pink using the ‘Color Picker’, the background color is still black as before.
The problem is solved when I’m changing the selected color from ‘first’ to ‘second’ and then back to ‘first’. Then the background color is pink as I expected.

I'd be happy to help resolve the issue with your assistance and guidance. 

**Steps to reproduce the behavior:**
1. Create two knobs - 'Background' using select and 'Color Picker' using color
2. Go to 'Background' knob
3. Choose 'first' as the background color
4. Go to 'Color Picker' knob
5. Change 'first' color from black to a different color
6. Background color does not changed, still black

**Expected behavior:**
1. Create two knobs - 'Background' using select and 'Color Picker' using color 
2. Go to 'Background' knob
3. Choose 'first' as the background color
4. Go to 'Color Picker' knob
5. Change 'first' color from black to a different color, for example pink
6. Background color changes to pink

  System:
    OS: macOS Mojave 10.14.5
    CPU: (8) x64 Intel(R) Core(TM) i7-8569U CPU @ 2.80GHz
  Binaries:
    Node: 12.13.1 - /usr/local/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.12.1 - /usr/local/bin/npm
  Browsers:
    Chrome: 80.0.3987.162
    Safari: 12.1.1

**attached screenshots:**
![Screen_Shot_2020-04-01_at_16 20 38](https://user-images.githubusercontent.com/51264062/78237302-ce8d8200-74e3-11ea-9fee-754e770c02d8.png)
![Screen_Shot_2020-04-01_at_16 21 16](https://user-images.githubusercontent.com/51264062/78237314-d2210900-74e3-11ea-8e5f-28e7de26a7d0.png)

**Comment by ByeBye-Sama** [comment]
I have pretty much the same behavior when in one knob I choose the color and then the other one where you can change properties of the color selected, but the option selected by default that is the color you choose doesn't change until you change manually the knob. I hope you can fix it. Thanks for storybook 🎉

**Comment by shilman** [comment]
Hi gang, We’ve just released addon-controls in 6.0-beta!

Controls are portable, auto-generated knobs that are intended to replace addon-knobs long term.

- [Controls README](https://github.com/storybookjs/storybook/tree/next/addons/controls)
- [Migration instructions](https://github.com/storybookjs/storybook/blob/next/addons/controls/README.md#how-do-i-migrate-from-addon-knobs)

Please upgrade and try them out today. Thanks for your help and support getting this stable for release!

**Comment by shilman** [comment]
For anybody who is interested in Controls but don't know where to start, I've created a quick & dirty step-by-step walkthrough to go from a fresh CRA project to a working demo. Check it out:

=> [Storybook Controls w/ CRA & TypeScript](https://gist.github.com/shilman/69c1dd41a466bae137cc88bd2c6ef487)

There are also some "knobs to controls" migration docs in the Controls README:

=> [How do I migrate from addon-knobs?](https://github.com/storybookjs/storybook/blob/next/addons/controls/README.md#how-do-i-migrate-from-addon-knobs)
